examples/registry/push_pull_copy.go

```diff
@@ -0,0 +1,192 @@
+// This example shows how to use oras go library to do some
```

Comment:
Consider splitting these up into smaller example files which target only one scenarios. 
For e.g. a download manifest should be a few lines of code in the file.

Reply:
Since the scenarios have dependencies on each other (e.g. pulling a blob requires pushing it first, resolving a tag requires pushing an tagging first), we still need to put all the registry-related scenarios in a single file, otherwise there will be to much redundant code.

For each scenario, I have added ad-hoc function in the `function.go` file. If a user need to copy some sample snippet to implement their own go app, they can copy the function directly.

If you have any comment or concern, please let me know.